pkg/runtimeproxy/server/docker/handler.go

```diff
@@ -156,131 +155,104 @@ func (d *RuntimeManagerDockerServer) HandleCreateContainer(ctx context.Context,
 	}
 }
 
+func (d *RuntimeManagerDockerServer) parseContainerInfo(url string) (*store.ContainerInfo, string, error) {
+	// we need to get the container id, because we need it to get info from checkpoint
+	containerID, err := getContainerID(url)
+	if err != nil {
+		return nil, "", err
+	}
+	// TODO: currently to check container is sandbox/container by checking container id existence in local-store
+	return store.GetContainerInfo(containerID), containerID, nil
+}
+
 func (d *RuntimeManagerDockerServer) HandleStartContainer(ctx context.Context, wr http.ResponseWriter, req *http.Request) {
 	// we need to get the container id, because we need it to get info from checkpoint
-	containerID, err := getContainerID(req.URL.Path)
+	containerMeta, _, err := d.parseContainerInfo(req.URL.Path)
 	if err != nil {
-		klog.Errorf("Failed to get container id, err: %v", err)
-		http.Error(wr, err.Error(), http.StatusInternalServerError)
+		d.generateErrorInfo(wr, fmt.Errorf("failed to get container id, err: %v", err))
 		return
 	}
-	containerMeta := store.GetContainerInfo(containerID)
-	runtimeHookPath := config.NoneRuntimeHookPath
-	var hookReq interface{}
-	if containerMeta != nil {
-		runtimeHookPath = config.StartContainer
-		hookReq = containerMeta.GetContainerResourceHookRequest()
-	}
 
 	// no need to care about the resp
-	if _, err := d.dispatcher.Dispatch(ctx, runtimeHookPath, config.PreHook, hookReq); err != nil {
-		klog.Errorf("Failed to call pre start container hook server %v", err)
-		http.Error(wr, err.Error(), http.StatusInternalServerError)
+	if _, err := d.dispatcher.Dispatch(ctx, config.StartContainer, config.PreHook, containerMeta.GetContainerResourceHookRequest()); err != nil {
```

Comment:
if containerMeta == nil (sandbox container),  we cant't send it to StartContainer hook point.